src/ast.ts

```diff
@@ -381,16 +406,17 @@ export class Symbols {
 
     if (this[type][name]) {
       this.log.error("Duplicated " + type + " definition: " + name + ".");
-      return;
+    } else {
+      this[type][name] = object;
     }
-    this[type][name] = object;
+    return this[type][name];
   }
 
-  registerFunction(name: string, params: string[], body: Exp) {
-    this.register('functions', name, method(params, body));
+  registerFunction(name: string, params: string[], body: Exp): Method {
```

Comment:
I wonder if register needs to be so broad in scope or whether registerFunction etc. for all the types would be better. I have not looked at its usage yet though.